crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -0,0 +1,705 @@
+                message,
+                nonce,
+                notary_public_key,
+            )
+            .await?;
+        let engine_toolkit_receipt = transaction_preview
+            .radix_engine_toolkit_receipt
+            .ok_or(CommonError::FailedToExtractTransactionReceiptBytes)?;
+
+        // Analyze the manifest
+        let execution_summary = transaction_manifest
+            .execution_summary_with_engine_toolkit_receipt(
+                engine_toolkit_receipt,
+            )?;
+
+        // Transactions created outside of the Wallet are not allowed to use reserved instructions
+        if !is_wallet_transaction
+            && !execution_summary.reserved_instructions.is_empty()
+        {
+            return Err(
+                CommonError::ReservedInstructionsNotAllowedInManifest {
+                    reserved_instructions: execution_summary
+                        .reserved_instructions
+                        .iter()
+                        .map(|i| i.to_string())
+                        .collect(),
+                },
+            );
+        }
+
+        Ok(TransactionToReview {
+            transaction_manifest,
+            execution_summary,
+        })
+    }
+
+    async fn get_transaction_preview(
+        &self,
+        gateway_client: GatewayClient,
+        manifest: TransactionManifest,
+        network_id: NetworkID,
+        message: Message,
+        nonce: Nonce,
+        notary_public_key: PublicKey,
+    ) -> Result<TransactionPreviewResponse> {
+        let signer_public_keys =
+            self.extract_transaction_signers(manifest.summary()).await?;
+        let epoch = gateway_client.current_epoch().await?;
+        let header = TransactionHeader::new(
+            network_id,
+            epoch,
+            Epoch::from(epoch.0 + 10),
+            nonce,
+            notary_public_key,
+            signer_public_keys.is_empty(),
+            0,
+        );
+        let intent = TransactionIntent::new(header, manifest, message)?;
+        let request = TransactionPreviewRequest::new(
+            intent,
+            signer_public_keys,
+            TransactionPreviewRequestFlags::new(true, false, false),
+        );
+        let response = gateway_client.transaction_preview(request).await?;
+        if response.receipt.status != TransactionReceiptStatus::Succeeded {
+            return Err(Self::map_failed_transaction_preview(response));
+        };
+        Ok(response)
+    }
+
+    fn map_failed_transaction_preview(
+        response: TransactionPreviewResponse,
+    ) -> CommonError {
+        let message = response
+            .receipt
+            .error_message
+            .unwrap_or_else(|| "Unknown reason".to_string());
+
+        // Quite rudimentary, but it is not worth making something smarter,
+        // as the GW will provide in the future strongly typed errors
+        let is_failure_due_to_deposit_rules = message
+            .contains("AccountError(DepositIsDisallowed")
```

Comment:
right, I would at the very list make these String Literals into const variables. can be global and non public

```
/// PLEASE DOCUMENT ME
const GW_ERR_ACCOUNT_DEPOSIT_DISALLOWED: &str = "AccountError(DepositIsDisallowed";

/// PLEASE DOCUMENT ME
const GW_... = "AccountError(NotAllBucketsCouldBeDeposited";
```